src/object/cli_obj.c

```diff
@@ -424,6 +438,23 @@ obj_grp_leader_get(struct dc_object *obj, int idx, unsigned int map_ver)
 	return rc;
 }
 
+/* If the client has been asked to fetch (list/query) from leader replica,
+ * then means that related data is associated with some prepared DTX that
+ * may be committable on the leader replica. According to our current DTX
+ * batched commit policy, it is quite possible that such DTX is not ready
+ * to be committed, or it is committable but cached on the leader replica
+ * for some time. On the other hand, such DTX may contain more data update
+ * than current fetch. If the subsequent fetch against the same redundancy
+ * group come very soon (within the OBJ_FETCH_LEADER_INTERVAL), then it is
+ * possible that related target for the next fetch is convered by the same
+ * DTX that is still not committed yet. If the assumption is right, asking
+ * the application to fetch from leader replica directly can avoid one RPC
+ * round-trip with non-leader replica. If such assumption is wrong, it may
+ * increase the server load on which the leader replica resides in a short
+ * time but it will not correctness issues.
+ */
+#define		OBJ_FETCH_LEADER_INTERVAL	3
```

Comment:
3 seconds is too coarse? I would suggest to set it to 1 second.

Reply:
Currently, the default batched commit interval is 60 seconds, plus the schedule delay, 1 second age may be too short and help very limited. Let's set it as 2 seconds.